src/overlays/actors/ovl_En_Mag/z_en_mag.c

```diff
@@ -27,21 +33,553 @@ const ActorInit En_Mag_InitVars = {
+    Gfx* gfx = *gfxp;
+
+    gDPLoadMultiBlock_4b(gfx++, maskTex, 0x0000, 0, G_IM_FMT_I, maskWidth, maskHeight, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                         G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+    if (1) {}
+
+    if (!flag) {
+        gDPLoadMultiBlock(gfx++, effectTex, 0x0100, 1, G_IM_FMT_I, G_IM_SIZ_8b, effectWidth, effectHeight, 0,
+                          G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, 5, shifts, shiftt);
+
+        gDPSetTileSize(gfx++, 1, 0, this->effectScroll & 0x7F, 0x7C, (this->effectScroll & 0x7F) + 0x7C);
+    }
+
+    gSPTextureRectangle(gfx++, rectLeft * 4, rectTop * 4, (rectLeft + rectWidth) * 4, (rectTop + rectHeight) * 4,
+                        G_TX_RENDERTILE, 0, 0, dsdx, dtdy);
+
+    *gfxp = gfx;
+}
+
+void EnMag_DrawImageRGBA32(Gfx** gfxp, s16 startLeft, s16 startTop, u8* source, u32 sizeX, u32 sizeY) {
+    Gfx* gfx = *gfxp;
+    u8* curTexture;
+    s32 textureCount;
+    u32 rectLeft;
+    u32 rectTop;
+    u32 height;
+    s32 remainingSize;
+    s32 textureSize;
+    s32 pad;
+    s32 i;
+
+    func_80094D28(&gfx);
+
+    curTexture = source;
+    rectLeft = startLeft - (sizeX >> 1);
+    rectTop = startTop - (sizeY >> 1);
+    height = 4096 / (sizeX << 2);
+    remainingSize = (sizeX * sizeY) << 2;
+    textureSize = (sizeX * height) << 2;
+    textureCount = remainingSize / textureSize;
+    if ((remainingSize % textureSize) != 0) {
+        textureCount++;
+    }
+
+    gSetTileCustom(gfx++, G_IM_FMT_RGBA, G_IM_SIZ_32b, sizeX, height, 0, G_TX_NOMIRROR | G_TX_CLAMP,
+                   G_TX_NOMIRROR | G_TX_CLAMP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+    remainingSize -= textureSize;
+
+    for (i = 0; i < textureCount; i++) {
+        gDPSetTextureImage(gfx++, G_IM_FMT_RGBA, G_IM_SIZ_32b, sizeX, curTexture);
+
+        gDPLoadSync(gfx++);
+        gDPLoadTile(gfx++, G_TX_LOADTILE, 0, 0, (sizeX - 1) << 2, (height - 1) << 2);
+
+        gSPTextureRectangle(gfx++, rectLeft << 2, rectTop << 2, (rectLeft + (s32)sizeX) << 2, (rectTop + height) << 2,
+                            G_TX_RENDERTILE, 0, 0, 1024, 1024);
+
+        if (1) {}
+
+        curTexture += textureSize;
+        rectTop += height;
+
+        if ((remainingSize - textureSize) < 0) {
+            if (remainingSize > 0) {
+                height = remainingSize / (s32)(sizeX << 2);
+                remainingSize -= textureSize;
+
+                gSetTileCustom(gfx++, G_IM_FMT_RGBA, G_IM_SIZ_32b, sizeX, height, 0, G_TX_NOMIRROR | G_TX_CLAMP,
+                               G_TX_NOMIRROR | G_TX_CLAMP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+            }
+        } else {
+            remainingSize -= textureSize;
+        }
+    }
+
+    *gfxp = gfx;
+}
+
+void EnMag_DrawTextFont(Gfx** gfxp, u8* texture, s32 rectLeft, s32 rectTop) {
```

Comment:
This seems clearly dedicated to drawing the text font characters, otherwise it should take arguments instead of using YREGs and hardcoded width/height imo. And since it's strictly local, I think there is no need to account for a more generic usage.

For now I'll rename the function to `DrawCharTexture` based on your suggestion, to make it more explicit.